### PR TITLE
Minor: Add `Column::from(Tableref, &FieldRef)`, `Expr::from(Column)` and `Expr::from(Tableref, &FieldRef)`

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/datafusion/pull/9595 from @haohuaijin 

## Rationale for this change

@appletreeisyellow and I noticed this while upgrading IOx -- there are times when the code would like to create `Expr::Column` that reference each of the columns in a schema

To do so it would be nice to be able to create a `Column` directly from a `&FieldRef` rather than having to to first call `.as_ref()` on the `FieldRef

## What changes are included in this PR?

Changes:
1. Add `Column::from(Tableref, &FieldRef)`
2. Add `Expr::from(Column)`
3. Add `Expr::from(Tableref, &FieldRef)`
4. Change some code to use the new API
5. Add example

## Are these changes tested?
By existing tests

### CODE REVIEW

File: datafusion/common/src/column.rs [hunk 1]
```diff
@@ -346,6 +348,13 @@ impl From<(Option<&TableReference>, &Field)> for Column {
     }
 }
 
+/// Create a column, use qualifier and field name
+impl From<(Option<&TableReference>, &FieldRef)> for Column {
```
Comment (alamb) [inline datafusion/common/src/column.rs:352]: This is the new API -- the rest of this PR updates the code to use it when possible

File: benchmarks/src/tpch/convert.rs [hunk 2]
```diff
@@ -88,9 +88,8 @@ impl ConvertOpt {
                 .schema()
                 .iter()
                 .take(schema.fields.len() - 1)
-                .map(|(qualifier, field)| {
-                    Expr::Column(Column::from((qualifier, field.as_ref())))
-                })
+                .map(Column::from)
```
Comment (comphead) [inline benchmarks/src/tpch/convert.rs:91]: can we still use a single map?

Comment (alamb) [reply benchmarks/src/tpch/convert.rs]: Good call -- I added some more `Expr::from` impls in  c4bc7e36d and now this looks pretty nice

File: benchmarks/src/tpch/convert.rs [hunk 3]
```diff
@@ -88,9 +88,7 @@ impl ConvertOpt {
                 .schema()
                 .iter()
                 .take(schema.fields.len() - 1)
-                .map(|(qualifier, field)| {
-                    Expr::Column(Column::from((qualifier, field.as_ref())))
-                })
+                .map(Expr::from)
```
Comment (alamb) [inline benchmarks/src/tpch/convert.rs:91]: This is now pretty nice I think given @comphead 's suggestion ❤️

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb
CIL testdoc keeps failing

**Comment by alamb** [comment]
> lgtm thanks @alamb CIL testdoc keeps failing

Thanks -- fixed in 4db07ee31

**Review by andygrove — APPROVED** [review]
Thanks @alamb

**Review by haohuaijin — APPROVED** [review]
LGTM, thank you @alamb

**Comment by alamb** [comment]
Thanks everyone for the reviews ❤️

**Pull request merged**